modules/SofaGraphComponent/SceneCheckerVisitor.h

```diff
@@ -40,9 +42,22 @@ class SOFA_GRAPH_COMPONENT_API SceneCheckerVisitor : public Visitor
     virtual ~SceneCheckerVisitor() ;
 
     void validate(Node* node) ;
+
+    void enableValidationAPIVersion(Node *node, const std::string& currentApiLevel) ;
+    void enableValidationRequiredPlugins(Node* node) ;
+
     virtual Result processNodeTopDown(Node* node) override ;
+
+    void installChangeSets() ;
+    void addHookInChangeSet(const std::string& version, ChangeSetHookFunction fct) ;
 private:
     std::map<std::string,bool> m_requiredPlugins ;
+    bool m_isRequiredPluginValidationEnabled {true} ;
+    bool m_isAPIVersionValidationEnabled {true} ;
+    std::string m_currentApiLevel {"17.12"} ;
+    std::string m_selectedApiLevel {"17.06"} ;
```

Comment:
Could we use `SOFAFRAMEWORK_VERSION` or `SOFA_VERSION` or `SOFA_VERSION_STR` instead to avoid hard coded default versions? See https://github.com/sofa-framework/sofa/blob/master/SofaKernel/SofaFramework/CMakeLists.txt#L172-L179